scilpy/tractograms/streamline_operations.py

```diff
@@ -854,3 +854,30 @@ def get_streamlines_bounding_box(streamlines):
         box_max = np.maximum(box_max, np.max(s, axis=0))
 
     return box_min, box_max
+
+
+def get_streamlines_as_fixed_array(streamlines):
+    """
+    Obtain streamlines as a fixed array of shape
+    (nbr of streamline, max streamline length, 3).
+
+    Useful for optimization with code precompiling. (See Numba)
+
+    Parameters
+    ----------
+    streamlines: list
+
+    Return
+    ------
+    streamlines_fixed: ndarray
+        Streamlines as a fixed array, padded with 0.
+    lengths: list
+        List of all the streamline lengths.
+    """
+    lengths = [len(streamline) for streamline in streamlines]
+    streamlines_fixed = np.zeros((len(streamlines), max(lengths), 3))
```

Comment:
if you have the lengths you can use it to find the first point of each streamline. and you won't need to zero-pad each stremlines shorter than the max_lengths.